src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -482,15 +428,45 @@ private void ProcessRenderQueue()
             }
         }
 
-        private void InvokeRenderCompletedCalls(ArrayRange<RenderTreeDiff> updatedComponents)
+        private Task InvokeRenderCompletedCalls(ArrayRange<RenderTreeDiff> updatedComponents)
```

Comment:
Changed this from `async void` to `Task` returning which we ignore. This follows the same pattern we would tell users to do, plus avoids any state machine if everything is synchronous.